### PR TITLE
Pass token expiration in LoginUserResponse

### LINKED ISSUE #274: Add token expiration of LoginUser response
Right now clients have no way of knowing when an access token will expire.
The only way to know if a token has expired, is to:

1. Make a request to Encryptonize
1. Get an Unauthenticated response back
1. Login again
1. Retry the original request

Making the expiration information available somehow, fx. in the LoginUser response, would allow clients to know when the token should be refreshed.

### PR DESCRIPTION
**Pull request by kim-cc**
### Description
Passing the token expiration time simplifies the client, as it can check if the token should be refreshed

### Relevant Issues/PRs
issue #274 

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [X] Added tests
- [ ] Refactor
- [ ] Bugfix
- [ ] Updated documentation
- [ ] Kubernetes changes
- [ ] CI/CD workflow changes

### Testing Checklist
- [ ] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [X] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [X] I have pulled in the latest changes from master.
- [X] I have run `make lint`.
- [X] I have successfully run `make tests`.
- [ ] I have updated relevant CI/CD workflows.
- [ ] I have updated relevant Kubernetes manifests/scripts.
- [ ] I have updated/added relevant (internal and external) documentation (readme, doc comments, etc).
- [ ] I have updated the dependency map to reflect my changes.
- [ ] I have added the license to new source code files.
- [X] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: client/client_test.go [hunk 1]
```diff
@@ -42,6 +43,8 @@ func TestMain(m *testing.M) {
 		endpoint = value
 	}
 
+	fmt.Printf("UID: %v, Pass: %v\n", uid, password)
```
Comment (philip-cc) [inline client/client_test.go:46]: This doesn't seem like a good idea.

Comment (kim-cc) [reply client/client_test.go:46]: Ahh yes, some debug output I forgot to delete. Fixed.

File: client/client.go [hunk 2]
```diff
@@ -31,11 +33,12 @@ import (
...
-	refClient  *grpcreflect.Client
-	ctx        context.Context
-	reflSource grpcurl.DescriptorSource
-	authHeader []string
+	connection      *grpc.ClientConn
+	refClient       *grpcreflect.Client
+	ctx             context.Context
+	reflSource      grpcurl.DescriptorSource
+	authHeader      []string
+	tokenExpiration time.Time
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply client/client.go:41]: Yes that would probably be a good idea. I have added a new method

File: encryption-service/common/access_token.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package common
+
+import "time"
+
+type AccessToken struct {
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply encryption-service/common/access_token.go]: Should be done now

File: encryption-service/impl/authn/token.go [hunk 4]
```diff
@@ -28,21 +28,21 @@ import (
 
 var ErrTokenExpired = errors.New("token expired")
 
-// AccessToken is the internal representation of an access token
-type AccessToken struct {
+// AccessTokenInfo is the internal representation of an access token
+type AccessTokenInfo struct {
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply encryption-service/impl/authn/token.go]: Should be done now

File: encryption-service/common/access_token.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+	Token      string
+	ExpiryTime time.Time
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply encryption-service/common/access_token.go]: Added

**Review by philip-cc — DISMISSED** [review]

File: encryption-service/impl/crypt/cryptor.go [hunk 6]
```diff
@@ -17,9 +17,8 @@ import (
 	"bytes"
 	"crypto/rand"
 	"encoding/gob"
+	"encryption-service/common"
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply encryption-service/impl/crypt/cryptor.go]: I have done this all the places I could find now

**Review by philip-cc — APPROVED** [review]

**Pull request merged**